src/components/DatasetPreview.vue

```diff
@@ -1,26 +1,103 @@
 <template>
     <div class="py-3">
         <h3>{{ dataset.name }}</h3>
-        <span v-html="this.description"></span>
+        <div v-if="!debug" >
+        <v-chip v-for="term in availableAnnotationsIncludedTerms" 
+                :key="term.termUri" @click="handleChipClick(term)" 
+                small :color="getChipColor(term.objectClass)">
+                {{ term.termName }} 
+                <v-icon right>mdi-plus</v-icon>
+          </v-chip> 
+        </div>
+        <div v-else >
+        <v-chip v-for="term in includedTerms" 
+                :key="term.termUri" @click="handleChipClick(term)" 
+                small :color="getChipColor(term.objectClass)" :outlined="!availableAnnotationsIncludedTerms.includes(term)">
+                {{ term.termName }} 
+                <v-icon right>mdi-plus</v-icon>
+          </v-chip> 
+        </div>
+
+        <div v-html="this.description"></div>
     </div>
 </template>
 
 <script>
 import { highlight } from "@/search-utils";
-import { marked } from "@/config/gemma";
+import { marked, axiosInst, baseUrl, excludedTerms, excludedCategories } from "@/config/gemma";
+import { mapState } from "vuex";
 
 export default {
   name: "DatasetPreview",
   props: {
-    dataset: Object
+    dataset: Object,
+    availableAnnotations: Array    
+  },
+  data() {
+    return {
+      terms: [],
+      includedTerms: []
+    }
   },
   computed: {
     description() {
       if (this.dataset.searchResult !== undefined && this.dataset.searchResult.highlights !== null && "description" in this.dataset.searchResult.highlights) {
         return marked.parseInline(highlight(this.dataset.description, this.dataset.searchResult.highlights.description));
       }
       return marked.parseInline(this.dataset.description);
+    },
+    chipColorMap() {
+      return {
+        FactorValue: 'yellow',
+        ExperimentTag: 'green',
+        BioMaterial: 'blue'
+      };
+    },
+    availableAnnotationsIncludedTerms() {
+      return this.includedTerms.filter(term => {
+        // Check if the termUri is included in any of the children arrays
+        return this.availableAnnotations.some(annotation =>
+          annotation.children.some(child => child.termUri === term.termUri)
```

Comment:
Use `getId()` here to compare terms you have with those from the collection. It will allow you to handle category and free-text terms.

Also, you should build a set of IDs to make the code more efficient and then filter in a single traversal.

Reply:
commit 25367abcae516cfadd696c8196abd195c3279492